### ISSUE TITLE
User model login endpoint: security issue with possible query injection

### ISSUE THREAD

**Issue by gabjauf** [strongloop/loopback#4195]
Hello Loopback!

I found some kind of vulnerability on POST /Users/login route from the default User model:

# Description/Steps to reproduce

Send the following to /Users/login as credentials
```js
{
	"email": {"neq": "foo" },
	"password": "anything you want"
}
```

When inspecting the resulting user here https://github.com/strongloop/loopback/blob/4b3a3a347c9176f1a86b00d229fc942bdc60c794/common/models/user.js#L254 , this results in the first user in the database being returned, which means we now only need to find the password.

Found in loopback v3.25.1

# Expected result

We would expect the email to be forced as a string.

Feel free to reach me if my description is unclear 😄

**Comment by gabjauf** [comment]
We may also use the "regexp" functionality to test multiple users without having to know their exact email:
{
	"email": {"regexp": "^a" },
	"password": "anything you want"
}

**Comment by hacksparrow** [comment]
@gabjauf I tried both the payloads, I am getting 401. Can you elaborate the steps?

**Comment by hacksparrow** [comment]
Oh, I got what you mean. Taking a look. Thanks for reporting.

**Comment by fabien** [comment]
@hacksparrow I was seeing a 401 as well - could you elaborate on 'I got what you mean'?

**Comment by bajtos** [comment]
The fix has been published last week in `loopback@3.26.0`.

**Comment by gabjauf** [comment]
Great works guys, thanks 👍
